File: FinanceApp.Shared/Dto/Loan/CreateLoan.cs
using FinanceApp.Shared.Dto.Base;

namespace FinanceApp.Shared.Dto.Loan
{
    public class CreateLoan : CreateDto
    {
        public string Name { get; set; }
        public DateTime InitialDate { get; set; }
        public int MonthsPayment { get; set; }
        public double LoanValue { get; set; }
        public double InterestRate { get; set; }
        public int Type { get; set; }

    }
}

File: FlutterApp/finance_app/lib/models/loan/create_loan.dart
class CreateLoan {
  String name;
  DateTime initialDate;
  int monthsPayment;
  double loanValue;
  double interestRate;
  int type;

  CreateLoan({
    required this.name,
    required this.initialDate,
    required this.monthsPayment,
    required this.loanValue,
    required this.interestRate,
    required this.type,
  });

  Map<String, dynamic> toJson() => {
        'name': name,
        'initialDate': initialDate.toString(),
        'monthsPayment': monthsPayment,
        'loanValue': loanValue,
        'interestRate': interestRate,
        'type': type,
      };
}
